client/packages/common/src/hooks/useNativeClient/useNativeClient.ts

```diff
@@ -88,6 +90,26 @@ export const useNativeClient = ({
     return result.log || noResult;
   };
 
+  const allowSleep = () => {
+    // Currently only supported on native platforms via capacitor
+    if (!Capacitor.isNativePlatform) return;
+
+    (async () => {
+      const result = await KeepAwake.isSupported();
+      if (result.isSupported) await KeepAwake.allowSleep();
+    })();
```

Comment:
what is the reason for the floating promise? or why not wait for `isSupported` and `allowSleep` to return? there are are some potential race condition when calling these methods concurrently...

Reply:
only because they are called from synchronous code, and it makes the usage clumsy if this method is async. I've refactored in 60669044b to wait for the promise to return before acting. Thanks!